api/handlers.go

```diff
@@ -890,8 +890,47 @@ func getEvents(c *context, w http.ResponseWriter, r *http.Request) {
 		f.Flush()
 	}
 
-	c.eventsHandler.Add(r.RemoteAddr, w)
-	c.eventsHandler.Wait(r.RemoteAddr, until)
+	eventsChan, cancelFunc := c.watchQueue.Watch()
+	defer cancelFunc()
+
+	// create timer for --until
+	timer := time.NewTimer(0)
```

Comment:
We shouldn't create the timer in the case where `--until` is not specified. It would fire immediately, and the `select` would always be able to take the `<-timer.C` branch, hogging CPU.

How about something like this:

```go
var (
        timer *time.Timer
        timerCh <-chan time.Time
)
if until > 0 {
        dur := time.Unix(until, 0).Sub(time.Now())
        timer = time.NewTimer(dur)
        timerCh = timer.C
}
```

Then have a `case <-timerCh:` in the select below.